### PR TITLE
Allow enabling worker threads on dev server

### PR DESCRIPTION
**Pull request by afbarbaro**
See this issue for context on use case.  [/node-clinic/issues/260](https://github.com/clinicjs/node-clinic/issues/260).

There's already an experimental config option to turn on worker threads but it's not being passed to the worker instantiated on the dev server.

### CODE REVIEW

**Comment by afbarbaro** [comment]
https://github.com/jlord/sheetsee.js/issues/26

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | afbarbaro/next.js enable-worker-threads-dev-server | Change |
| - | - | - | - |
| buildDuration | 12s | 11.9s | -63ms |
| nodeModulesSize | 42.8 MB | 42.8 MB | ⚠️ +232 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | afbarbaro/next.js enable-worker-threads-dev-server | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.229 | 2.198 | -0.03 |
| / avg req/sec | 1121.74 | 1137.39 | +15.65 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.464 | 1.5 | ⚠️ +0.04 |
| /error-in-render avg req/sec | 1707.67 | 1666.75 | ⚠️ -40.92 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | afbarbaro/next.js enable-worker-threads-dev-server | Change |
| - | - | - | - |
| 677f882d2ed8..a2e7.js gzip | 13.4 kB | 13.4 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-HASH.js gzip | 6.65 kB | 6.65 kB | ✓ |
| webpack-HASH.js gzip | 751 B | 751 B | ✓ |
| Overall change | 59.8 kB | 59.8 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | afbarbaro/next.js enable-worker-threads-dev-server | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31.3 kB | 31.3 kB | ✓ |
| Overall change | 31.3 kB | 31.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | afbarbaro/next.js enable-worker-threads-dev-server | Change |
| - | - | - | - |
| _app-fde3324..9dd1.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-af59f..582f.js gzip | 3.46 kB | 3.46 kB | ✓ |
| amp-9716187d..0aa8.js gzip | 536 B | 536 B | ✓ |
| hooks-107e90..74c7.js gzip | 888 B | 888 B | ✓ |
| index-ac435c..ecf2.js gzip | 227 B | 227 B | ✓ |
| link-c0d2c96..de48.js gzip | 1.67 kB | 1.67 kB | ✓ |
| routerDirect..dc9d.js gzip | 303 B | 303 B | ✓ |
| withRouter-6..0e02.js gzip | 302 B | 302 B | ✓ |
| Overall change | 8.66 kB | 8.66 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | afbarbaro/next.js enable-worker-threads-dev-server | Change |
| - | - | - | - |
| _buildManifest.js gzip | 347 B | 347 B | ✓ |
| Overall change | 347 B | 347 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | afbarbaro/next.js enable-worker-threads-dev-server | Change |
| - | - | - | - |
| index.html gzip | 612 B | 612 B | ✓ |
| link.html gzip | 620 B | 620 B | ✓ |
| withRouter.html gzip | 608 B | 608 B | ✓ |
| Overall change | 1.84 kB | 1.84 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | afbarbaro/next.js enable-worker-threads-dev-server | Change |
| - | - | - | - |
| buildDuration | 13.9s | 13.4s | -493ms |
| nodeModulesSize | 42.8 MB | 42.8 MB | ⚠️ +232 B |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | afbarbaro/next.js enable-worker-threads-dev-server | Change |
| - | - | - | - |
| 677f882d2ed8..a2e7.js gzip | 13.4 kB | 13.4 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-HASH.js gzip | 6.65 kB | 6.65 kB | ✓ |
| webpack-HASH.js gzip | 751 B | 751 B | ✓ |
| Overall change | 59.8 kB | 59.8 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | afbarbaro/next.js enable-worker-threads-dev-server | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31.3 kB | 31.3 kB | ✓ |
| Overall change | 31.3 kB | 31.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | afbarbaro/next.js enable-worker-threads-dev-server | Change |
| - | - | - | - |
| _app-fde3324..9dd1.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-af59f..582f.js gzip | 3.46 kB | 3.46 kB | ✓ |
| amp-9716187d..0aa8.js gzip | 536 B | 536 B | ✓ |
| hooks-107e90..74c7.js gzip | 888 B | 888 B | ✓ |
| index-ac435c..ecf2.js gzip | 227 B | 227 B | ✓ |
| link-c0d2c96..de48.js gzip | 1.67 kB | 1.67 kB | ✓ |
| routerDirect..dc9d.js gzip | 303 B | 303 B | ✓ |
| withRouter-6..0e02.js gzip | 302 B | 302 B | ✓ |
| Overall change | 8.66 kB | 8.66 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | afbarbaro/next.js enable-worker-threads-dev-server | Change |
| - | - | - | - |
| _buildManifest.js gzip | 347 B | 347 B | ✓ |
| Overall change | 347 B | 347 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | afbarbaro/next.js enable-worker-threads-dev-server | Change |
| - | - | - | - |
| _error.js | 1.02 MB | 1.02 MB | ✓ |
| 404.html | 2.67 kB | 2.67 kB | ✓ |
| 500.html | 2.65 kB | 2.65 kB | ✓ |
| amp.amp.html | 10.6 kB | 10.6 kB | ✓ |
| amp.html | 1.86 kB | 1.86 kB | ✓ |
| hooks.html | 1.92 kB | 1.92 kB | ✓ |
| index.js | 1.02 MB | 1.02 MB | ✓ |
| link.js | 1.08 MB | 1.08 MB | ✓ |
| routerDirect.js | 1.07 MB | 1.07 MB | ✓ |
| withRouter.js | 1.07 MB | 1.07 MB | ✓ |
| Overall change | 5.27 MB | 5.27 MB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Webpack 5 Mode</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | afbarbaro/next.js enable-worker-threads-dev-server | Change |
| - | - | - | - |
| buildDuration | 14s | 14.1s | ⚠️ +49ms |
| nodeModulesSize | 42.8 MB | 42.8 MB | ⚠️ +232 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | afbarbaro/next.js enable-worker-threads-dev-server | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.237 | 2.232 | 0 |
| / avg req/sec | 1117.36 | 1120.01 | +2.65 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.482 | 1.527 | ⚠️ +0.04 |
| /error-in-render avg req/sec | 1686.79 | 1637.28 | ⚠️ -49.51 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | afbarbaro/next.js enable-worker-threads-dev-server | Change |
| - | - | - | - |
| 597-2bc2376a..203d.js gzip | 13.3 kB | 13.3 kB | ✓ |
| framework.HASH.js gzip | 39.3 kB | 39.3 kB | ✓ |
| main-HASH.js gzip | 6.59 kB | 6.59 kB | ✓ |
| webpack-HASH.js gzip | 954 B | 954 B | ✓ |
| Overall change | 60.2 kB | 60.2 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | afbarbaro/next.js enable-worker-threads-dev-server | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31.1 kB | 31.1 kB | ✓ |
| Overall change | 31.1 kB | 31.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | afbarbaro/next.js enable-worker-threads-dev-server | Change |
| - | - | - | - |
| _app-0c62a59..94b7.js gzip | 1.26 kB | 1.26 kB | ✓ |
| _error-97d24..ed28.js gzip | 3.38 kB | 3.38 kB | ✓ |
| amp-2926e4c2..9ccc.js gzip | 536 B | 536 B | ✓ |
| hooks-1ed65b..8908.js gzip | 902 B | 902 B | ✓ |
| index-6259b6..77d8.js gzip | 230 B | 230 B | ✓ |
| link-b722682..14a4.js gzip | 1.65 kB | 1.65 kB | ✓ |
| routerDirect..862a.js gzip | 306 B | 306 B | ✓ |
| withRouter-4..76fd.js gzip | 302 B | 302 B | ✓ |
| Overall change | 8.57 kB | 8.57 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | afbarbaro/next.js enable-worker-threads-dev-server | Change |
| - | - | - | - |
| _buildManifest.js gzip | 322 B | 322 B | ✓ |
| Overall change | 322 B | 322 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | afbarbaro/next.js enable-worker-threads-dev-server | Change |
| - | - | - | - |
| index.html gzip | 586 B | 586 B | ✓ |
| link.html gzip | 593 B | 593 B | ✓ |
| withRouter.html gzip | 579 B | 579 B | ✓ |
| Overall change | 1.76 kB | 1.76 kB | ✓ |

</details>

#### Diffs
<details>
<summary>Diff for <strong>index.html</strong></summary>

```diff
@@ -43,7 +43,7 @@
         "props": { "pageProps": {} },
         "page": "/",
         "query": {},
-        "buildId": "pWKHDys1boG-Ynr-uSBlX",
+        "buildId": "D_-lh34HM4zsrYbxZ9HxF",
         "isFallback": false,
         "gip": true
       }
@@ -77,11 +77,11 @@
       async=""
     ></script>
     <script
-      src="/_next/static/pWKHDys1boG-Ynr-uSBlX/_buildManifest.js"
+      src="/_next/static/D_-lh34HM4zsrYbxZ9HxF/_buildManifest.js"
       async=""
     ></script>
     <script
-      src="/_next/static/pWKHDys1boG-Ynr-uSBlX/_ssgManifest.js"
+      src="/_next/static/D_-lh34HM4zsrYbxZ9HxF/_ssgManifest.js"
       async=""
     ></script>
   </body>
```
</details>
<details>
<summary>Diff for <strong>link.html</strong></summary>

```diff
@@ -48,7 +48,7 @@
         "props": { "pageProps": {} },
         "page": "/link",
         "query": {},
-        "buildId": "pWKHDys1boG-Ynr-uSBlX",
+        "buildId": "D_-lh34HM4zsrYbxZ9HxF",
         "isFallback": false,
         "gip": true
       }
@@ -82,11 +82,11 @@
       async=""
     ></script>
     <script
-      src="/_next/static/pWKHDys1boG-Ynr-uSBlX/_buildManifest.js"
+      src="/_next/static/D_-lh34HM4zsrYbxZ9HxF/_buildManifest.js"
       async=""
     ></script>
     <script
-      src="/_next/static/pWKHDys1boG-Ynr-uSBlX/_ssgManifest.js"
+      src="/_next/static/D_-lh34HM4zsrYbxZ9HxF/_ssgManifest.js"
       async=""
     ></script>
   </body>
```
</details>
<details>
<summary>Diff for <strong>withRouter.html</strong></summary>

```diff
@@ -43,7 +43,7 @@
         "props": { "pageProps": {} },
         "page": "/withRouter",
         "query": {},
-        "buildId": "pWKHDys1boG-Ynr-uSBlX",
+        "buildId": "D_-lh34HM4zsrYbxZ9HxF",
         "isFallback": false,
         "gip": true
       }
@@ -77,11 +77,11 @@
       async=""
     ></script>
     <script
-      src="/_next/static/pWKHDys1boG-Ynr-uSBlX/_buildManifest.js"
+      src="/_next/static/D_-lh34HM4zsrYbxZ9HxF/_buildManifest.js"
       async=""
     ></script>
     <script
-      src="/_next/static/pWKHDys1boG-Ynr-uSBlX/_ssgManifest.js"
+      src="/_next/static/D_-lh34HM4zsrYbxZ9HxF/_ssgManifest.js"
       async=""
     ></script>
   </body>
```
</details>
</details>
Commit: a70017b9d48e4f6e72b4ef2ab8e0527be0a267b7

File: packages/next/server/next-dev-server.ts [hunk 1]
```diff
@@ -104,6 +104,7 @@ export default class DevServer extends Server {
       {
         maxRetries: 0,
         numWorkers: this.nextConfig.experimental.cpus,
+        enableWorkerThreads: this.nextConfig.experimental.workerThreads,
         forkOptions: {
           env: {
             ...process.env,
```
Comment (almoshaks69) [inline packages/next/server/next-dev-server.ts:109]: Dd

**Review by almoshaks69 — DISMISSED** [review]

**Review by aleee1007 — DISMISSED** [review]

**Review by timneutkens — COMMENTED** [review]
This one intentionally does not use worker threads because of incompatibility with some Node.js libraries, maybe that's fine though as it's opt-in

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Build</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | afbarbaro/next.js enable-worker-threads-dev-server | Change |
| - | - | - | - |
| buildDuration | 11.9s | 12.2s | ⚠️ +321ms |
| buildDurationCached | 3s | 2.7s | -289ms |
| nodeModulesSize | 49.5 MB | 49.5 MB | ⚠️ +194 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | afbarbaro/next.js enable-worker-threads-dev-server | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.076 | 2.028 | -0.05 |
| / avg req/sec | 1204.21 | 1232.72 | +28.51 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.111 | 1.117 | ⚠️ +0.01 |
| /error-in-render avg req/sec | 2250.34 | 2238.38 | ⚠️ -11.96 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | afbarbaro/next.js enable-worker-threads-dev-server | Change |
| - | - | - | - |
| 359.HASH.js gzip | 2.96 kB | 2.96 kB | ✓ |
| 745.HASH.js gzip | 180 B | 180 B | ✓ |
| framework-HASH.js gzip | 42.2 kB | 42.2 kB | ✓ |
| main-HASH.js gzip | 20.9 kB | 20.9 kB | ✓ |
| webpack-HASH.js gzip | 1.53 kB | 1.53 kB | ✓ |
| Overall change | 67.8 kB | 67.8 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | afbarbaro/next.js enable-worker-threads-dev-server | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31.1 kB | 31.1 kB | ✓ |
| Overall change | 31.1 kB | 31.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | afbarbaro/next.js enable-worker-threads-dev-server | Change |
| - | - | - | - |
| _app-HASH.js gzip | 803 B | 803 B | ✓ |
| _error-HASH.js gzip | 3.06 kB | 3.06 kB | ✓ |
| amp-HASH.js gzip | 554 B | 554 B | ✓ |
| css-HASH.js gzip | 329 B | 329 B | ✓ |
| dynamic-HASH.js gzip | 2.52 kB | 2.52 kB | ✓ |
| head-HASH.js gzip | 2.28 kB | 2.28 kB | ✓ |
| hooks-HASH.js gzip | 903 B | 903 B | ✓ |
| image-HASH.js gzip | 5.61 kB | 5.61 kB | ✓ |
| index-HASH.js gzip | 261 B | 261 B | ✓ |
| link-HASH.js gzip | 1.66 kB | 1.66 kB | ✓ |
| routerDirect..HASH.js gzip | 319 B | 319 B | ✓ |
| script-HASH.js gzip | 387 B | 387 B | ✓ |
| withRouter-HASH.js gzip | 320 B | 320 B | ✓ |
| bb14e60e810b..30f.css gzip | 125 B | 125 B | ✓ |
| Overall change | 19.1 kB | 19.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | afbarbaro/next.js enable-worker-threads-dev-server | Change |
| - | - | - | - |
| _buildManifest.js gzip | 490 B | 490 B | ✓ |
| Overall change | 490 B | 490 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | afbarbaro/next.js enable-worker-threads-dev-server | Change |
| - | - | - | - |
| index.html gzip | 531 B | 531 B | ✓ |
| link.html gzip | 544 B | 544 B | ✓ |
| withRouter.html gzip | 525 B | 525 B | ✓ |
| Overall change | 1.6 kB | 1.6 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Webpack 4 Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | afbarbaro/next.js enable-worker-threads-dev-server | Change |
| - | - | - | - |
| buildDuration | 9.9s | 9.7s | -178ms |
| buildDurationCached | 3.7s | 4s | ⚠️ +317ms |
| nodeModulesSize | 49.5 MB | 49.5 MB | ⚠️ +194 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | afbarbaro/next.js enable-worker-threads-dev-server | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.082 | 2.119 | ⚠️ +0.04 |
| / avg req/sec | 1200.84 | 1179.54 | ⚠️ -21.3 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.112 | 1.078 | -0.03 |
| /error-in-render avg req/sec | 2248.38 | 2319.82 | +71.44 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | afbarbaro/next.js enable-worker-threads-dev-server | Change |
| - | - | - | - |
| 17.HASH.js gzip | 2.98 kB | 2.98 kB | ✓ |
| 18.HASH.js gzip | 185 B | 185 B | ✓ |
| 677f882d2ed8..HASH.js gzip | 13.7 kB | 13.7 kB | ✓ |
| framework.HASH.js gzip | 41.9 kB | 41.9 kB | ✓ |
| main-HASH.js gzip | 8.4 kB | 8.4 kB | ✓ |
| webpack-HASH.js gzip | 1.22 kB | 1.22 kB | ✓ |
| Overall change | 68.4 kB | 68.4 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | afbarbaro/next.js enable-worker-threads-dev-server | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31.3 kB | 31.3 kB | ✓ |
| Overall change | 31.3 kB | 31.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | afbarbaro/next.js enable-worker-threads-dev-server | Change |
| - | - | - | - |
| _app-HASH.js gzip | 791 B | 791 B | ✓ |
| _error-HASH.js gzip | 3.76 kB | 3.76 kB | ✓ |
| amp-HASH.js gzip | 552 B | 552 B | ✓ |
| css-HASH.js gzip | 333 B | 333 B | ✓ |
| dynamic-HASH.js gzip | 2.71 kB | 2.71 kB | ✓ |
| head-HASH.js gzip | 2.97 kB | 2.97 kB | ✓ |
| hooks-HASH.js gzip | 911 B | 911 B | ✓ |
| index-HASH.js gzip | 231 B | 231 B | ✓ |
| link-HASH.js gzip | 1.64 kB | 1.64 kB | ✓ |
| routerDirect..HASH.js gzip | 298 B | 298 B | ✓ |
| script-HASH.js gzip | 3.02 kB | 3.02 kB | ✓ |
| withRouter-HASH.js gzip | 294 B | 294 B | ✓ |
| e025d2764813..52f.css gzip | 125 B | 125 B | ✓ |
| Overall change | 17.6 kB | 17.6 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | afbarbaro/next.js enable-worker-threads-dev-server | Change |
| - | - | - | - |
| _buildManifest.js gzip | 500 B | 500 B | ✓ |
| Overall change | 500 B | 500 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | afbarbaro/next.js enable-worker-threads-dev-server | Change |
| - | - | - | - |
| index.html gzip | 577 B | 577 B | ✓ |
| link.html gzip | 587 B | 587 B | ✓ |
| withRouter.html gzip | 569 B | 569 B | ✓ |
| Overall change | 1.73 kB | 1.73 kB | ✓ |

</details>

</details>
Commit: 5f112c76145149cada2b9b00ae96152de5ad2355

**Comment by ijjk** [comment]
## Failing test suites 
Commit: 5f112c76145149cada2b9b00ae96152de5ad2355

**test/integration/route-load-cancel/test/index.test.js** 
- next/dynamic > dev mode > should cancel slow page loads on re-navigation

<details>
<summary>Expand output</summary>

  ● next/dynamic › dev mode › should cancel slow page loads on re-navigation

    NoSuchElementError: no such element: Unable to locate element: {"method":"css selector","selector":"#page-text"}
      (Session info: headless chrome=91.0.4472.114)

      31 |     await waitFor(1000)
      32 |
    > 33 |     const text = await browser.elementByCss('#page-text').text()
         |                  ^
      34 |     expect(text).toMatch(/2/)
      35 |     expect(await browser.eval('window.routeCancelled')).toBe('yes')
      36 |   })

      at Object.throwDecodedError (../node_modules/selenium-webdriver/lib/error.js:550:15)
      at parseHttpResponse (../node_modules/selenium-webdriver/lib/http.js:565:13)
      at Executor.execute (../node_modules/selenium-webdriver/lib/http.js:491:26)
      at thenableWebDriverProxy.execute (../node_modules/selenium-webdriver/lib/webdriver.js:700:17)
      at Object.<anonymous> (integration/route-load-cancel/test/index.test.js:33:18)

</details>

**Review by ijjk — APPROVED** [review]
This change looks good as it matches our other `Worker` usage during build and is opt-in as mentioned by Tim, thanks for the PR!

**Pull request merged**